File: solutions/c/resistor-color-duo/1/resistor_color_duo.c
#include "resistor_color_duo.h"

/*typedef enum {
  BLACK,
  BROWN,
  RED,
  ORANGE,
  YELLOW,
  GREEN,
  BLUE,
  VIOLET,
  GREY,
  WHITE
} resistor_band_t;*/

uint16_t color_code(resistor_band_t bands[]){
  int res = (int)bands[0] * 10;
  res += (int)bands[1];
  return res;
}

File: solutions/csharp/resistor-color-duo/1/ResistorColorDuo.cs
enum ResistorColor{
    Black,
    Brown,
    Red,
    Orange,
    Yellow,
    Green,
    Blue,
    Violet,
    Grey,
    White
}

public static class ResistorColorDuo
{
    public static int Value(string[] colors)
    {
        int res = (int)Enum.Parse(typeof(ResistorColor), colors[1], true);
        res += 10 * (int)Enum.Parse(typeof(ResistorColor), colors[0], true);
        return res;
    }
}
